adapters/repos/db/lsmkv/segment_group_cleanup.go

```diff
@@ -351,7 +355,12 @@ func (c *segmentCleanerCommon) readEarliestCleaned(ids, sizes []int64, nowTs int
 					// not the left one, it is unknown what was last segment used for cleanup of removed
 					// left segment, therefore compacted segment will be cleaned again using all newer segments.
 					possibleStartIdx := idx + 1
+					// in case of using segments that were already used for cleanup, process them in reverse
+					// order starting with newest ones, to maximize the chance of finding redundant entries
+					// as soon as possible (leaving segments that were already used for cleanup as last ones)
```

Comment:
**praise:** this does make sense. So the logic here is that probably the new tombstones will be present in the newer segments if we older ones have been cleaned up before already. 
** thought: ** On the longer term, we could have some metrics/logs describing how many entries were deleted on a cleanup and how many were merged on a compaction.